js/ui/codeView.js

```diff
@@ -672,17 +659,25 @@ var CodingSession = new Lang.Class({
         this._appActionProxy = null;
     },
 
-    removeAppWindow: function() {
-        // Save the actor, so we can complete the destroy transition later
-        if (this.appRemovedByFlipBack)
-            this._appRemovedActor = this.app;
+    removeFlippedBackAppWindow: function() {
+        if (!this.appRemovedByFlipBack)
+            return false;
 
-        this.appRemovedByFlipBack = false;
+        // Save the actor, so we can complete the destroy transition later
+        this._appRemovedActor = this.app;
         this.app = null;
+        this.appRemovedByFlipBack = false;
 
+        // We assume we have a toolbox here, or this.appRemovedByFlipBack would be false
         this._setState(STATE_TOOLBOX);
 
-        this._completeRemoveWindow();
+        let actor = this._actorForCurrentState();
```

Comment:
I see that all uses of this function verify whether `actor` is `null`, is that needed elsewhere/here?